presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -519,37 +576,14 @@ private void destroyIfNecessary()
                             driverContext.getTaskId());
                 }
             }
-            if (driverContext.getMemoryUsage() > 0) {
-                log.error("Driver still has memory reserved after freeing all operator memory.");
-            }
-            if (driverContext.getSystemMemoryUsage() > 0) {
-                log.error("Driver still has system memory reserved after freeing all operator memory.");
-            }
-            if (driverContext.getRevocableMemoryUsage() > 0) {
-                log.error("Driver still has revocable memory reserved after freeing all operator memory. Freeing it.");
-            }
-            driverContext.finished();
-        }
-        catch (Throwable t) {
-            // this shouldn't happen but be safe
-            inFlightException = addSuppressedException(
-                    inFlightException,
-                    t,
-                    "Error destroying driver for task %s",
-                    driverContext.getTaskId());
         }
         finally {
             // reset the interrupted flag
             if (wasInterrupted) {
                 Thread.currentThread().interrupt();
             }
         }
-
-        if (inFlightException != null) {
-            // this will always be an Error or Runtime
-            Throwables.throwIfUnchecked(inFlightException);
-            throw new RuntimeException(inFlightException);
-        }
+        return inFlightException;
```

Comment:
Should we check that memory usage in operatorContext is 0? OperatorContext does not currently have a `getMemoryUsage`, but it's very straightforward to add one.